internal/pkg/skuba/kubernetes/node_version_test.go

```diff
@@ -18,394 +18,1066 @@
+			}
+		})
+	}
+}
+
+func TestNodeVersionInfoForClusterVersion(t *testing.T) {
+	tests := []struct {
+		node corev1.Node
+	}{
+		{
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: master0,
+					Labels: map[string]string{
+						kubeadmconstants.LabelNodeRoleMaster: "",
+					},
+				},
+			},
+		},
+		{
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: worker0,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+
+		var ver string
+		isMaster := IsControlPlane(&tt.node)
+		clusterVersions := StaticVersionInquirer{}.AvailablePlatformVersions()
+		for _, cv := range clusterVersions {
+			tName := fmt.Sprintf("%v node version info when cluster version is %v", tt.node.Name, cv)
+			t.Run(tName, func(t *testing.T) {
+				vInfo := StaticVersionInquirer{}.NodeVersionInfoForClusterVersion(&tt.node, cv)
+				if vInfo.Nodename == "" {
+					t.Error("node name expected, but none returned")
+				}
+				ver = vInfo.ContainerRuntimeVersion.String()
+				if _, err := version.ParseSemantic(ver); err != nil {
+					t.Errorf("container runtime version(%v) should parse semantic", ver)
+				}
+				ver = vInfo.KubeletVersion.String()
+				if _, err := version.ParseSemantic(ver); err != nil {
+					t.Errorf("kubelet version(%v) should parse semantic", ver)
+				}
+
+				if isMaster {
+					ver = vInfo.APIServerVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("api server version(%v) should parse semantic", ver)
+					}
+					ver = vInfo.ControllerManagerVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("controller manager version(%v) should parse semantic", ver)
+					}
+					ver = vInfo.SchedulerVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("scheduler version(%v) should parse semantic", ver)
+					}
+					ver = vInfo.EtcdVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("etcd version(%v) should parse semantic", ver)
+					}
+				}
+			})
+		}
+	}
+}
+
+func TestString(t *testing.T) {
+	cluster := "1.1.1"
+	tests := []struct {
+		name             string
+		isMaster         bool
+		apiServerVersion *version.Version
+		kubeletVersion   *version.Version
+	}{
+		{
+			name:             "master version",
+			isMaster:         true,
+			apiServerVersion: version.MustParseSemantic(cluster),
+		},
+		{
+			name:           "woker version",
+			kubeletVersion: version.MustParseSemantic(cluster),
+		},
+	}
+	clusterVersion := version.MustParseSemantic(cluster)
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			var nodeVersionInfo NodeVersionInfo
+			switch tt.isMaster {
+			case true:
+				nodeVersionInfo = NodeVersionInfo{
+					APIServerVersion: clusterVersion,
+				}
+			default:
+				nodeVersionInfo = NodeVersionInfo{
+					KubeletVersion: clusterVersion,
+				}
+			}
```

Comment:
It looks you want to test different nodeVersionInfo of node types. I would suggest remove this duplicated logic then move the different nodeVersionInfo as test data like below.

```
	tests := []struct {
		name             string
		nodeVersionInfo  *NodeVersionInfo
	}{
		{
			name:             "master version",
			nodeVersionInfo: &NodeVersionInfo{
					APIServerVersion: clusterVersion,
			}
		},
		{
			name:           "woker version",
			nodeVersionInfo: &NodeVersionInfo{
					KubeletVersion: clusterVersion,
				}
		},
	}
```